src/Ng/CommandHelpers.cs

```diff
@@ -324,99 +328,78 @@ public static StorageFactory CreateSuffixedStorageFactory(string suffix, IDictio
             {
                 throw new ArgumentNullException("suffix");
             }
-
-            Uri storageBaseAddress = null;
-            string storageBaseAddressStr;
-            if (arguments.TryGetValue("-storageBaseAddress" + suffix, out storageBaseAddressStr))
+            try
             {
-                storageBaseAddressStr = storageBaseAddressStr.TrimEnd('/') + "/";
+                Uri storageBaseAddress = null;
+                string storageBaseAddressStr;
+                TryGetArgument(arguments, Constants.StorageBaseAddress + suffix, out storageBaseAddressStr);
+                if (!string.IsNullOrEmpty(storageBaseAddressStr))
+                {
+                    storageBaseAddressStr = storageBaseAddressStr.TrimEnd('/') + "/";
 
-                storageBaseAddress = new Uri(storageBaseAddressStr);
-            }
+                    storageBaseAddress = new Uri(storageBaseAddressStr);
+                }
 
-            string storageType;
-            if (!arguments.TryGetValue("-storageType" + suffix, out storageType))
-            {
-                TraceRequiredArgument("-storageType" + suffix);
-                return null;
-            }
+                string storageType;
+                TryGetArgument(arguments, Constants.StorageType, out storageType, required: true);
 
-            if (storageType.Equals("File", StringComparison.InvariantCultureIgnoreCase))
-            {
-                string storagePath;
-                if (!arguments.TryGetValue("-storagePath" + suffix, out storagePath))
+                if (storageType.Equals("File", StringComparison.InvariantCultureIgnoreCase))
                 {
-                    TraceRequiredArgument("-storagePath" + suffix);
-                    return null;
-                }
+                    string storagePath;
+                    TryGetArgument(arguments, Constants.StoragePath + suffix, out storagePath, required: true);
 
-                if (storageBaseAddress == null)
-                {
-                    TraceRequiredArgument("-storageBaseAddress" + suffix);
-                    return null;
-                }
+                    if (storageBaseAddress == null)
+                    {
+                        TraceRequiredArgument("-storageBaseAddress" + suffix);
+                        return null;
+                    }
```

Comment:
storageBaseAddress is not the var checked directly above. it is reused and required for File type, but not for Azure type. So we null check here

---
In reply to: [71208452](https://github.com/NuGet/NuGet.Services.Metadata/pull/77#discussion_r71208452) [](ancestors = 71208452)